src/core/analysis/mcmc/MarginalLikelihoodEstimator.cpp

```diff
@@ -97,10 +97,19 @@ MarginalLikelihoodEstimator::MarginalLikelihoodEstimator(const path &fn, const s
             
                 continue;
             }
-        
+
+            // check for broken lines
+            if( columns.size() <= powerColumnIndex || columns.size() <= likelihoodColumnIndex ) 
+                throw RbException() << "Please check format of file " << fn << ", missing power and likelihood columns in some lines";
+
+            double p, l;
             // get the power entry
             std::string tmp = columns[powerColumnIndex];
-            double p = atof( tmp.c_str() );
+            try {
+                p = std::stod( tmp.c_str() );
```

Comment:
Doesn't std::stod takes a std::string as input instead of a c-array of characters?